### PR TITLE
implemented unlock method and added test cases

### PR DESCRIPTION
**Pull request by puneet-khushwani-eth**

### CODE REVIEW

File: test/Account/unlock.test.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    accountObject = new Account(accountCreationResponse.address, accountCreationResponse.encryptedKeyStore);
+  });
+
+  it('should unlock successfully with valid password', () => {
+    assert.strictEqual(
+      accountObject.unlock(web3, validPassword),
+      true,
+      'should return true for successful unlock',
+    );
+  });
```
Comment (deepesh-kn) [inline test/Account/unlock.test.ts]: Please spy and check if the following methods were called.
```
 const web3Account = web3.eth.accounts.decrypt(this.encryptedKeyStore, password);
web3.eth.accounts.wallet.add(web3Account);
```

File: test/Account/unlock.test.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
+import * as sinon from 'sinon';
+import { assert } from 'chai';
+
+import Account from '../../src/Account';
+import SpyAssert from '../utils/SpyAssert';
+
+const Web3 = require('web3');
+
+describe('Unlock', () => {
+  let web3: any;
```
Comment (abhayks1) [inline test/Account/unlock.test.ts]:
```suggestion
  let web3: Web3;
```

File: test/Account/unlock.test.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    validPassword = 'validPassword';
+    accountCreationResponse = Account.create(web3, validPassword);
+    accountObject = new Account(
+      accountCreationResponse.address,
+      accountCreationResponse.encryptedKeyStore
+    );
+  });
+
+  it('should unlock successfully with valid password', () => {
+    const dummyAccount = web3.eth.accounts.create();
```
Comment (abhayks1) [inline test/Account/unlock.test.ts]:
```suggestion
    const dummyAccount = await web3.eth.accounts.create();
```

File: src/Account.ts [hunk 4]
```diff
@@ -43,9 +43,16 @@ export default class Account {
    * @param password The password required to unlock the account.
    * @returns `true` if its unlocked otherwise false.
    */
-  // public unlock(web3: Web3, password: string): boolean {
-  //   // Unlocking the account and adding it to the local web3 instance so that everything is signed
-  //   // locally when using web3.eth.send
-  //   return false;
-  // }
+  public unlock(web3: Web3, password: string): boolean {
```
Comment (abhayks1) [inline src/Account.ts]: Why this function is not integrated / called from anywhere else?

File: test/Account/unlock.test.ts (hunk 2 above)
Comment (puneet-khushwani-eth) [reply test/Account/unlock.test.ts]: We are downgrading version of web to beta 37 which does not have type script support for all types. We are thus removing using Web3 types in our code

File: src/Account.ts (hunk 4 above)
Comment (puneet-khushwani-eth) [reply src/Account.ts]: The calls to this are in a separate PR.

File: test/Account/unlock.test.ts (hunk 3 above)
Comment (puneet-khushwani-eth) [reply test/Account/unlock.test.ts]: This is a sync command as per https://web3js.readthedocs.io/en/1.0/web3-eth-accounts.html.

**Pull request merged**